src/freenet/client/async/SplitFileFetcherStorage.java

```diff
@@ -1144,6 +1149,9 @@ public boolean run(ClientContext context) {
                     if(succeeded) return false;
                     succeeded = true;
                 }
+                synchronized(cooldownLock) {
+                    overallCooldownWakeupTime = -1;
+                }
                 fetcher.onSuccess();
```

Comment:
Is it okay to set overallCooldownWakeupTime to completed before calling `onSuccess()` / `fail()` / `failOnDiskError()` like this? I'd be concerned about something seeing the request as completed before the success / failure is marked. Does that not happen, or is it not a problem, and why?